Sources/StreamVideoSwiftUI/Utils/PictureInPicture/StreamPictureInPictureVideoRenderer.swift

```diff
@@ -113,45 +113,49 @@ final class StreamPictureInPictureVideoRenderer: UIView, RTCVideoRenderer {
     // MARK: - Rendering lifecycle
 
     /// This method is being called from WebRTC and asks the container to set its size to the track's size.
-    func setSize(_ size: CGSize) {
-        trackSize = size
+    nonisolated func setSize(_ size: CGSize) {
+        DispatchQueue.main.async {
+            self.trackSize = size
+        }
     }
 
-    func renderFrame(_ frame: RTCVideoFrame?) {
-        guard let frame = frame else {
-            return
-        }
+    nonisolated func renderFrame(_ frame: RTCVideoFrame?) {
+        DispatchQueue.main.async {
+            guard let frame else {
+                return
+            }
 
-        // Update the trackSize and re-calculate rendering properties if the size
-        // has changed.
-        trackSize = .init(width: Int(frame.width), height: Int(frame.height))
+            // Update the trackSize and re-calculate rendering properties if the size
+            // has changed.
+            self.trackSize = .init(width: Int(frame.width), height: Int(frame.height))
```

Comment:
Rather than strongly accessing self (and making many changes) use weak self in blocks and guard self. This is also ensuring no memory leaks while allowing for omitting self.

Reply:
fixed